### ISSUE TITLE
Multi-node cluster causes error creating indexes.

### ISSUE THREAD

**Issue by atul-tewari** [brantburnett/couchbase-index-manager#62]
Executing the following using 0.7.2 version against Couchbase 6.6.0 with multi-node cluster.

`./couchbase-index-manager -c couchbase://<ip of one of the nodes> -u Administrator -p <somepassword> sync -f <bucket> indexes.yaml `

`Error: An unknown N1QL error occured. This is usually related to an  out-of-memory condition. Check the errors responseBody property or inspect the cluster logs for further details.
    at ~/couchbase/node_modules/couchbase/lib/bucket.js:998:15
`
The above command works just fine against a single node cluster of Enterprise Couchbase 6.6.0.  Even a single index with the following YAML fails

```
---
name: idx_DocTypes
index_key:
  - docType
condition: (docType in ['Something1', 'Something2'])
num_replica: 1
```

As soon as another node is added to the cluster with both nodes using index services.

**Comment by brantburnett** [comment]
@atul-tewari 

So far I am not able to reproduce this problem, I'm wondering if it might be environmental. The steps I tried:

1. Use `env/Up.ps1 -Version 6.6.0` to spin up a 3 node cluster in Docker
2. Change `env/Exec.ps1` to use the image `btburnett3/couchbase-index-manager:0.7.2` instead of a local build
3. Run `env/Exec.ps1 -NoBuild sync -f beer-sample /example/beer-sample` to run the entire set of sample index definitions against the bucket

Is there any chance you can give me a more detailed set of steps to reproduce?

**Comment by atul-tewari** [comment]
Hello Brant,

Thank you for your response. After rebuilding the cluster again, I realized that the server names were using AWS’s private IPs to form the cluster.  The single node was 127.0.0.1 (localhost).

 

I am now setting up cluster where public DNS entries for each node in the cluster matches respective hostname of each node.  I used sdk-doctor to help me identify this issue.  I’m sorry I completely missed that.  I’ll follow-up confirming it works after I rebuild the cluster again.

 

Thank you so much for responding so quickly.

 

Atul
